va/http.go

```diff
@@ -414,18 +415,19 @@ func fallbackErr(err error) bool {
 	if err == nil {
 		return false
 	}
-
-	switch err := err.(type) {
-	case *url.Error:
+	var urlError *url.Error
+	var netOpError *net.OpError
+	if errors.As(err, &urlError) {
 		// URL Errors should be unwrapped and tested
-		return fallbackErr(err.Err)
-	case *net.OpError:
+		return fallbackErr(urlError.Err)
```

Comment:
We can actually use `errors.As` to get rid of this recursion case. If we directly do `errors.As(err, &netOpError)`, that will actually catch the case we care about, since `*url.Error` has an Unwrap method that returns `url.Err`.